go-controller/pkg/util/slice.go

```diff
@@ -0,0 +1,36 @@
+package util
+
+import "k8s.io/klog/v2"
+
+// RemoveIndexFromSliceUnstable attempts to remove slice index specified by parameter i. Slice order is not preserved.
+func RemoveIndexFromSliceUnstable[T comparable](slice []T, i int) []T {
+	var t T
+	sliceLen := len(slice)
+	if sliceLen == 0 {
+		klog.Warningf("Attempted to remove slice index %d from a slice of length 0", i)
+		return slice
+	}
+	if i >= sliceLen {
+		klog.Warningf("Attempted to remove slice index %d from a slice (%v) of length %d", i, slice, sliceLen)
+		return slice
+	}
+	if i < 0 {
+		klog.Warningf("Attempted to index slice with a negative number %d", i)
+		return slice
+	}
```

Comment:
I feel a bit uneasy logging from such a basic utility function.

I think I would just let it panic being such a basic function. Just as if we accessed an index out of bounds directly in any other part of the code, it is the user responsibility to ensure i is a valid index in `RemoveIndexFromSliceUnstable(slice, i)` just as it would be if he used `slice[i]`

Other alternative is to do  
```
if i >= sliceLen {
   return slice
}
```
and doc that if an index is out of bounds the original slice is returned. But I prefer this one less.

Reply:
Fair enough. I was swinging both ways in my head but you're right - consumers of this might miss a programming error due to the func input guardrails. Removed all input checks.